4/m_whoisgeolite.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2024 reverse
+ *
+ * This file contains a third-party module for InspIRCd. You can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/// $ModAuthor: reverse <mike.chevronnet@gmail.com>
+/// $ModDesc: Adds city information to WHOIS using the MaxMind database.
+/// $ModConfig: add path <geolite dbpath="path/geodata/GeoLite2-City.mmdb">
+/// $ModDepends: core 4
+
+/// $CompilerFlags: find_compiler_flags("libmaxminddb")
+/// $LinkerFlags: find_linker_flags("libmaxminddb")
+
+/// $PackageInfo: require_system("alpine") libmaxminddb-dev pkgconf
+/// $PackageInfo: require_system("arch") pkgconf libmaxminddb
+/// $PackageInfo: require_system("darwin") libmaxminddb pkg-config
+/// $PackageInfo: require_system("debian~") libmaxminddb-dev pkg-config
+/// $PackageInfo: require_system("rhel~") pkg-config libmaxminddb-devel
+
+#include "inspircd.h"
+#include "modules/whois.h"
+#include <maxminddb.h>
+
+class ModuleWhoisGeoLite final
+	: public Module
+	, public Whois::EventListener
+{
+private:
+	MMDB_s mmdb;       // MaxMind database object
+	bool db_loaded;    // Flag to indicate if the database was successfully loaded
+	std::string dbpath;
+
+public:
+	ModuleWhoisGeoLite()
+		: Module(VF_OPTCOMMON, "Adds city information to WHOIS using the MaxMind database.")
+		, Whois::EventListener(this), db_loaded(false)
+	{
+	}
+
+	void ReadConfig(ConfigStatus& status) override
+	{
+		// Load configuration and get the path to the GeoLite2 database
+		auto& tag = ServerInstance->Config->ConfValue("geolite");
+		dbpath = tag->getString("dbpath", "/etc/GeoLite2-City.mmdb");
+
+		// Attempt to open the MaxMind GeoLite2-City database
+		int status_open = MMDB_open(dbpath.c_str(), MMDB_MODE_MMAP, &mmdb);
+		if (status_open != MMDB_SUCCESS) {
+			ServerInstance->SNO.WriteGlobalSno('a', "GeoLite2: Failed to open GeoLite2 database: " + std::string(MMDB_strerror(status_open)));
```

Comment:
You probably just want to throw a ModuleException here because it doesn't make sense to have the module loaded without being configured. That will allow you to get rid of `db_loaded`.